pkg/combustion/registry.go

```diff
@@ -30,50 +28,48 @@ var haulerManifest string
 var registryScript string
 
 func configureRegistry(ctx *image.Context) ([]string, error) {
-	if IsEmbeddedArtifactRegistryEmpty(ctx.ImageDefinition.EmbeddedArtifactRegistry) {
+	if IsEmbeddedArtifactRegistryAndKubernetesManifestsEmpty(ctx) {
 		log.AuditComponentSkipped(registryComponentName)
 		return nil, nil
 	}
 
-	err := writeHaulerManifest(ctx)
-	if err != nil {
-		log.AuditComponentFailed(registryComponentName)
-		return nil, fmt.Errorf("writing hauler manifest: %w", err)
-	}
-
-	err = populateHaulerStore(ctx)
+	haulerBinaryPath := fmt.Sprintf("hauler-%s", string(ctx.ImageDefinition.Image.Arch))
+	err := copyHaulerBinary(ctx, haulerBinaryPath)
 	if err != nil {
 		log.AuditComponentFailed(registryComponentName)
-		return nil, fmt.Errorf("populating hauler store: %w", err)
+		return nil, fmt.Errorf("copying hauler binary: %w", err)
 	}
 
-	err = generateRegistryTar(ctx)
+	registryScriptNameResult, err := writeRegistryScript(ctx)
```

Comment:
This is the operation which should be the last in this component. All other steps are the prerequisites of having the final Combustion script written to the filesystem. This would also make templating the "registry" directory easier.